src/SUFST/src/can_unpack.c

```diff
@@ -151,5 +173,28 @@ void queue_receive_thread_entry(ULONG input)
         {
           return;
         }
+        /* For statistic */
+        unpack_ptr->stats.tx_pdu_count++;
+        unpack_ptr->stats.txbits += (sizeof(pdu_struct) * 8);     
     }
 }
+
+void stats_init(unpack_performance_t* stats)
+{
+    stats->rx_can_bps = 0;
+    stats->rx_can_count = 0;
+    stats->tx_pdu_bps = 0;
+    stats->tx_pdu_count = 0;
+    stats->rxbits = 0;
+    stats->txbits = 0;
+}
+
+void timer_callback(unpack_performance_t* stats)
```

Comment:
Is this function going to be thread safe? These operations are not guaranteed to be atomic. What happens if the unpack thread is currently running / changing the `rxbits` count and the timer finishes? Worth checking how ThreadX behaves in this scenario.

Might need a mutex to protect things.

Reply:
![image](https://user-images.githubusercontent.com/47193610/222435442-5572332b-a789-4eb4-a732-a84663f72777.png)
I think that means we don't have to use mutexes here.
[Source](https://learn.microsoft.com/en-us/azure/rtos/threadx/chapter3)

Reply:
I still think we do, think about the scenario:
- Unpack thread reads value of bytes received from RAM into register.
- Timer interrupt suspends unpack thread, context switch occurs and timer callback runs.
- Bytes received is set to zero during callback.
- Context switch back to unpack thread restores the old non-zero bytes received value, which is then incremented and stored.

This applies to all the statistics which are modified in the timer callback and unpack thread. Since incrementing / addition is not atomic, we can never guarantee that it will be safe.

Reply:
Oh okay, now agree. We can add mutexes in unpack thread to protect those operations.